spec/02-integration/20-wasm/08-declarative_spec.lua

```diff
@@ -0,0 +1,184 @@
+local helpers = require "spec.helpers"
+
+
+local function post_config(client, config)
+  config._format_version = config._format_version or "3.0"
+
+  local res = client:post("/config?flatten_errors=1", {
+    body = config,
+    headers = {
+      ["Content-Type"] = "application/json"
+    },
+  })
+
+  assert.response(res).has.jsonbody()
+
+  assert.logfile().has.no.line("[emerg]", true, 0)
+  assert.logfile().has.no.line("[crit]",  true, 0)
+  assert.logfile().has.no.line("[alert]", true, 0)
+  assert.logfile().has.no.line("[error]", true, 0)
+  assert.logfile().has.no.line("[warn]",  true, 0)
+
+  return res
+end
+
+
+local function expect_entity_error(res, err)
+  assert.response(res).has.status(400)
+
+  local json = assert.response(res).has.jsonbody()
+  assert.is_table(json.flattened_errors)
+
+  local found = false
+
+
+  for _, entity in ipairs(json.flattened_errors) do
+    assert.is_table(entity.errors)
+    for _, elem in ipairs(entity.errors) do
+      if elem.type == "entity" then
+        assert.same(err, elem.message)
+        found = true
+        break
+      end
+    end
+  end
+
+  assert.is_true(found, "expected '" .. err .. "' message in response")
+end
+
+
+describe("#wasm declarative config", function()
```

Comment:
This `08-declarative_spec.lua` only contains negative test cases. Shall we add a positive test case like the filter is successfully added for db-less? Or it already has been included in other test file?

Reply:
This is a good point. `05-cache-invalidation_spec.lua` _does_ test the same dbless API endpoint (`/config`), but it doesn't hurt to have at least a simple happy-path example in this file.

Reply:
Thanks for pointing out. Look good to me then.